### ISSUE TITLE
Commands in groups with similar names are not selected properly for execution.

### ISSUE THREAD

**Issue by james7132** [RogueException/Discord.Net#328]
I have a few commands/groups set up like so:

``` csharp
[Group("temp")]
public class Temp : ModuleBase {
  [Group("role")]
  public class Role : ModuleBase {
     [Command("add")]
     public async Task Add(...) {
     }
  }
}

[Group("role")]
public class Role : ModuleBase {
   [Command("add")]
   public async Task Add(...) {
   }
}
```

When trying to execute `role add`, `temp role add` is found and executed instead.

**Comment by FiniteReality** [comment]
The Command map appears to be registering commands twice, as seen here:
![](https://cdn.discordapp.com/attachments/81384956881809408/239866585412665345/unknown.png)

I will look into this further in the morning; it might be a bug from my command aliasing implementation which is causing this.

**Comment by RogueException** [comment]
Is this fixed by #361?

**Comment by james7132** [comment]
Seems to be resolved now.
